colabs/artifact_basics/Artifact_Basics.ipynb

```diff
@@ -0,0 +1,351 @@
+        "\n",
+        "\n",
+        "If you change or add any parameters names, be sure to replicate those changes in the following code samples as well."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bpGidixAICM1"
+      },
+      "source": [
+        "# Use an Artifact"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jZC69sSMdX4j"
+      },
+      "source": [
+        "When you need to select a particular Artifact, indicate the artifact you want to use with the `use_artifact` method. This returns a run object. In the proceeding code snippet specifies an artifact called `my_first_artifact` with the alias `latest`:\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7Ix_ciolJICe"
+      },
+      "outputs": [],
+      "source": [
+        "run = wandb.init(project=\"artifact-basics\")\n",
+        "artifact = run.use_artifact(artifact_or_name=\"my_first_artifact:latest\")\n",
+        "run.finish()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "u60Orwg5sYuI"
+      },
+      "source": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "# Create a new Artifact version"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "When you need add a new file, you can use the `artifact.add_file` method."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "run = wandb.init(project=\"artifact-basics\")\n",
+        "artifact = run.use_artifact(artifact_or_name=\"my_first_artifact:latest\") # selects the artifact you're adding the file to\n",
+        "artifact.add_file(local_path=\"/content/sample_data/california_housing_test.csv\", name=\"new_file\")\n",
+        "run.finish()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "This adds a new .csv file called `new_file` to the `my_first_artifact` Artifact. \n",
+        "\n",
+        "Optionally, you can choose to add an entire directory to an Artifact:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "run = wandb.init(project=\"artifact-basics\")\n",
+        "artifact = run.use_artifact(artifact_or_name=\"my_first_artifact:latest\") # selects the artifact you're adding the file to\n",
+        "artifact.add_dir(local_path=\"/content/sample_data\", name=\"new_directory\")\n",
+        "run.finish()"
```

Comment:
it looks like we're missing a run.log_artifact here

Reply:
one thing that still doesn't feel very clear is the fact that adding this sample_data dir replaces the contents in the previous version. At a high level, the workflow would look like:
1. I had a dataset that I added as a directory, i.e a directory with 100 images in it
2. I did some kind of pre-processing on it and removed 10 of the images in that directory (i.e I deleted them locally) because they were corrupt 
3. I again did add_dir on that directory
4. W&B knows that the contents of that directory changed so a new version is created under the hood to reflect the new contents of the directory (i.e w/o the 10 removed files)

Reply:
we should make it clear as well that the next version with have the alias latest and v1 (or more generally mention the alias scheme)